qiskit/circuit/quantumcircuit.py

```diff
@@ -106,113 +106,859 @@
+    constructor, or :meth:`copy_empty_like`.  However, it is also possible to add these afterwards.
+    Typed classical data, such as standalone :class:`~.expr.Var` nodes (see
+    :ref:`circuit-repr-real-time-classical`), can be both constructed and added with separate
+    methods.
+
+    New registerless :class:`.Qubit` and :class:`.Clbit` objects are added using :meth:`add_bits`.
+    These objects must not already be present in the circuit.  You can check if a bit exists in the
+    circuit already using :meth:`find_bit`.
 
-        name (str): the name of the quantum circuit. If not set, an
-            automatically generated string will be assigned.
-        global_phase (float or ParameterExpression): The global phase of the circuit in radians.
-        metadata (dict): Arbitrary key value metadata to associate with the
-            circuit. This gets stored as free-form data in a dict in the
-            :attr:`~qiskit.circuit.QuantumCircuit.metadata` attribute. It will
-            not be directly used in the circuit.
-        inputs: any variables to declare as ``input`` real-time variables for this circuit.  These
-            should already be existing :class:`.expr.Var` nodes that you build from somewhere else;
-            if you need to create the inputs as well, use :meth:`QuantumCircuit.add_input`.  The
-            variables given in this argument will be passed directly to :meth:`add_input`.  A
-            circuit cannot have both ``inputs`` and ``captures``.
-        captures: any variables that that this circuit scope should capture from a containing scope.
-            The variables given here will be passed directly to :meth:`add_capture`.  A circuit
-            cannot have both ``inputs`` and ``captures``.
-        declarations: any variables that this circuit should declare and initialize immediately.
-            You can order this input so that later declarations depend on earlier ones (including
-            inputs or captures). If you need to depend on values that will be computed later at
-            runtime, use :meth:`add_var` at an appropriate point in the circuit execution.
+    .. automethod:: add_bits
 
-            This argument is intended for convenient circuit initialization when you already have a
-            set of created variables.  The variables used here will be directly passed to
-            :meth:`add_var`, which you can use directly if this is the first time you are creating
-            the variable.
+    Registers are added to the circuit with :meth:`add_register`.  In this method, it is not an
+    error if some of the bits are already present in the circuit.  In this case, the register will
+    be an "alias" over the bits.  This is not generally well-supported by any hardware backends;
+    while it is valid in the OpenQASM 3 abstract data model, it is probably best to stay away from
+    relying on it.  The registers a given bit is in are part of the return of :meth:`find_bit`.
```

Comment:
I'm not sure we can make an absolute statement about this for all backends. I assume you were referring to IBM backends. Although the non-ibm backends I've worked with don't really model registers at all and just do the tracking in qiskit.

I also don't think we want to couple this with a discussion of the qasm3 data model because it's largely independent from that except if you're exporting the circuit to qasm3.

```suggestion
    be an "alias" over the bits.  This is not generally well-supported by most hardware backends;
    it is probably best to stay away from relying on it.  The registers a given bit is in are part of
     the return of :meth:`find_bit`.
```

Reply:
Done in 956475957f.